sdk/src/Services/S3/Generated/Model/Internal/MarshallTransformations/GetBucketMetricsConfigurationResponseUnmarshaller.cs

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ * 
+ *  http://aws.amazon.com/apache2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/*
+ * Do not modify this file. This file is generated from the s3-2006-03-01.normal.json service model.
+ */
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Xml.Serialization;
+
+using Amazon.S3.Model;
+using Amazon.Runtime;
+using Amazon.Runtime.Internal;
+using Amazon.Runtime.Internal.Transform;
+using Amazon.Runtime.Internal.Util;
+
+#pragma warning disable CS0612,CS0618
+namespace Amazon.S3.Model.Internal.MarshallTransformations
+{
+    /// <summary>
+    /// Response Unmarshaller for GetBucketMetricsConfiguration operation
+    /// </summary>  
+    public partial class GetBucketMetricsConfigurationResponseUnmarshaller : S3ReponseUnmarshaller
+    {
+        /// <summary>
+        /// Unmarshaller the response from the service to the response class.
+        /// </summary>  
+        /// <param name="context"></param>
+        /// <returns></returns>
+        public override AmazonWebServiceResponse Unmarshall(XmlUnmarshallerContext context)
+        {
+            GetBucketMetricsConfigurationResponse response = new GetBucketMetricsConfigurationResponse();
+            UnmarshallResult(context,response);
+            
+            PostUnmarshallCustomization(context, response);
+            return response;
+        }        
+
+        private static void UnmarshallResult(XmlUnmarshallerContext context, GetBucketMetricsConfigurationResponse response)
+        {
+            int originalDepth = context.CurrentDepth;
+            int targetDepth = originalDepth + 1;
```

Comment:
(Forgot to publish this one...)

Do we have an integration test for this operation? The manual code had a section I don't see here (or the updated metrics unmarshaller): https://github.com/aws/aws-sdk-net/blob/eb138204fbefd914b30bb84c9eedf4a228108065/sdk/src/Services/S3/Custom/Model/Internal/MarshallTransformations/GetBucketMetricsConfigurationResponseUnmarshaller.cs#L58-L59